app/models/user.rb

```diff
@@ -27,11 +27,6 @@ class User < ActiveRecord::Base
   # We really need a Dispatch Chain here or something.
   # This will also let us return a human error message.
   #
-  def self.authenticate(login, password)
-    return nil if login.blank? || password.blank?
-    u = find_by_login(login.downcase) # need to get the salt
-    u && u.authenticated?(password) ? u : nil
-  end
```

Comment:
This method is also not used anywhere else.
And it doesn't even work
it uses the `authenticated?` method which does not seem to exist
So I removed this as well

Reply:
have you used the grep command to check that these methods which you removed are not invoked anywhere?

Reply:
I used the silver searcher command to search for it.
Didn't find the method being called